temporal/fx.go

```diff
@@ -278,19 +274,25 @@ func (s ServerFx) Stop() error {
 	return s.app.Stop(context.Background())
 }
 
-func StopService(logger log.Logger, app *fx.App, svcName primitives.ServiceName, stopChan chan struct{}) {
-	stopCtx, cancelFunc := context.WithTimeout(context.Background(), serviceStopTimeout)
```

Comment:
Let's just use the context that fx already provides to stop hooks.

Reply:
Which context do you mean? This is the thing calling Stop, it's providing the context

Reply:
It's not immediately available here. We should propagate it, instead of ignoring it, here: https://github.com/temporalio/temporal/blob/89f2095bf8a258835391ea48e11a3e1e03af6ba3/temporal/fx.go#L869-L871

Reply:
Oh, you mean passing the context from the _outer_ App to the _inner_ Apps. Yeah, we can do that. Right now it's just Background, though, that outer context doesn't use a timeout. Which makes me think... should we put the timeout on that one instead? But passing the dynamic timeout up another level will be even more messy. That makes me lean towards either no timeout or a large fixed one and forget about this dynamic thing

Reply:
Wow, that was _much_ harder than I thought it would be. The problem is that the outer app wasn't actually using fx properly: it wasn't returning from Start (if configured with blocking Start, which we do in our main), and just internally stopping the inter apps from within Start, bypassing fx's Stop.

I refactored things, but it's big enough that I want to make a separate PR for it, let's leave this just for the shutdown timeout